src/llama.cpp

```diff
@@ -17190,6 +17190,7 @@ bool llama_model_has_encoder(const struct llama_model * model) {
     switch (model->arch) {
         case LLM_ARCH_T5:        return true;
         case LLM_ARCH_T5ENCODER: return true;
+        case LLM_ARCH_JINA_BERT_V2: return true;
```

Comment:
The `llama_model_has_encoder` is used to indicate if the model architecture contains an encoder part. Currently only T5 encoder-decoder models are of this kind. Jina models are decoder-only, so we should not add them in this list

I've pushed an alternative fix here: https://github.com/ggerganov/llama.cpp/pull/8997

Please give it a try and let me know

Reply:
🤔 I thought jina bert model are encoder only, thanks for #8997, it works!

Reply:
Hm, yes you are right. We are using `llama_decode` for BERT models which got me confused. Probably should be changed to `llama_encode` in the future. For now the #8997 should be OK